app/models/book.rb

```diff
@@ -7,6 +7,8 @@ class Book < ApplicationRecord
     :author_id,
     images_attributes: Image::IMAGE_PARAMS].freeze
 
+  acts_as_paranoid
```

Comment:
cái này hoạt động ntn em

Reply:
thay em xoá 1 record nào đó hẳn khỏi CSDL thì em sẽ giấu nó đi.  
Thì trong rails hỗ trợ gem "paranoia". Tạo thêm deleted_at  trong table mình cần dùng sort delete. Trường này sẽ lưu thời gian xoá record đó. nhiệm vụ của gem này e nghĩ là tạo ra database ảo để lưu cái record bị xoá

Reply:
db ảo là ảo ntn
e check xem nó update or insert gì trong DB nhé

Reply:
tại sao phải cần sort delete

Reply:
Ví dụ như một thẻ mượn của em có các cuốn sách, nhưng Admin lại xoá cái cuốn sách đó. Sau mà muốn lấy lại thông tin của các thẻ mượn đó thì cuốn sách bị Admin xoá sẽ không còn trong thẻ mượn đó nữa ạ. Vậy sẽ phải dùng sort delete để lưu thông tin cuốn sách đó để bảo toàn dữ liệu cho hệ thống. 
Khi xoá record nào đó, Dữ liệu không bị xoá khỏi CSDL mà chỉ  sẽ update thời gian xoá trong trường deleted_at.
Vậy gem "paranoia" em nghĩ nó sẽ giấu dữ liệu xoá. Khi  dùng object.all thì record đó sẽ không được hiển thị chị ạ